src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -683,10 +683,10 @@ private static double CreateDoubleNaN(bool sign, ulong significand)
         }
 
         private static float CreateSingle(bool sign, byte exp, uint sig)
-            => BitConverter.Int32BitsToSingle((int)(((sign ? 1U : 0U) << float.SignShift) | ((uint)exp << float.ExponentShift) | sig));
+            => BitConverter.Int32BitsToSingle((int)(((sign ? 1U : 0U) << float.SignShift) + ((uint)exp << float.ExponentShift) + sig));
```

Comment:
The issue was that this was doing a bitwise or when it should have been using addition.

Reply:
The original implementation was doing the correct thing: https://github.com/ucb-bar/berkeley-softfloat-3/blob/master/source/include/internals.h#L107